### PR TITLE
Issue 257 - "Quality-Diversity Generative Sampling for Learning with Synthetic Data"

### LINKED ISSUE #257: Quality-Diversity Generative Sampling for Learning with Synthetic Data
- Paper name: Quality-Diversity Generative Sampling for Learning with Synthetic Data
- ArXiv Link: https://arxiv.org/pdf/2312.14369

To close this issue open a PR with a paper report using the provided [report template](https://github.com/Dahoas/QDSyntheticData/blob/main/papers/paper_example.md).

### PR DESCRIPTION
**Pull request by lauraaisling**
Closes #257

### CODE REVIEW

File: papers/2312.14369_Quality-Diversity_Generative_Sampling_for_Learning_with_Synthetic_Data [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+## Summary
+
+This paper deals with the problem that generative models can create synthetic training datasets, but may transfer biases to downstream tasks. 
+It focuses on protecting quality and diversity when generating synthetic training datasets.
+They propose quality-diversity generative sampling (QDGS), a framework for using quality-diversity optimisation using prompt guidance, without fine-tuning the generative model. 
+It uses prompt guidance to sample balanced data from biased generative models, across a user-defined measure space. 
+QDGS increases spread over desired attributes, described by user-defined language prompts, by exploring high dimensionality latent spaces.
+
+Basically uses a latent space exploration algorithm to iteratively identify inputs for a generative model that produces an image.
```
Comment (Dahoas) [inline papers/2312.14369_Quality-Diversity_Generative_Sampling_for_Learning_with_Synthetic_Data:13]: Is the idea basically to run CMA-MAEGA (some evolutionary algorithm) in the latent space to produce a diverse set of images (note: I don't really know what CMA-MAEGA is)

**Comment by Dahoas** [comment]
This is a really interesting paper!

File: papers/2312.14369_Quality-Diversity_Generative_Sampling_for_Learning_with_Synthetic_Data (hunk 1 above)
Comment (Dahoas) [reply papers/2312.14369_Quality-Diversity_Generative_Sampling_for_Learning_with_Synthetic_Data:13]: Note: MAEGA - Map Elites via gradient aboresence (an update to map-elites)

Comment (Dahoas) [reply papers/2312.14369_Quality-Diversity_Generative_Sampling_for_Learning_with_Synthetic_Data:13]: Iteratively updates latent parameter by adding gradient of objective + gradient of measure function

Comment (lauraaisling) [reply papers/2312.14369_Quality-Diversity_Generative_Sampling_for_Learning_with_Synthetic_Data:13]: CMA-MAEGA is a QD optimisation algorithm combining the following 2 QD algorithms: 

- CMA-MA - Covariance matrix adaptation map-annealing [1]. 
It's a update to MAP Elites. 
-  MEGA - MAP-Elites via a Gradient Arborescence [2]. 
MEGA is a DQD (both the objective and measure functions are first-order differentiable) algorithm introduced in this paper. 

[1] "Covariance matrix adaptation map-annealing" Fontaine and Nikolaidis, '23. https://arxiv.org/abs/2205.10752 
[2] "Differentiable Quality Diversity" Fontaine and Nikolaidis, '21. https://arxiv.org/abs/2106.03894

Comment (lauraaisling) [reply papers/2312.14369_Quality-Diversity_Generative_Sampling_for_Learning_with_Synthetic_Data:13]: Searching over latent solutions is referred to as "latent space illumination". 
It's the problem of finding an input θ to a generative model, that produces diverse images across (a user defined) measure space. 

Section 3.3 has a formalism of the algorithm. 
Summary: 
Have objective function f for image generation.
We have k user defined measures (m_j), for each of these measures, 
We wish to find an input θ that maximises f and gives diverse measures. 
We map solutions for initial θ into the measure space. 
θ is updated to θ' with objective-measure gradients (See equation 1 and algorithm 1). 

The objective function f in their examples computes CLIP similarity scores of generated images with positive and negative language prompts (Equation 2). 
Each measure function m_j also computes CLIP scores to determine the measures of diversity in the resulting images.

In synthetic face example, the objective function f is a weighted sum of three component
functions: a text objective, an image margin, and a regularization factor. Measure dimensions, are found with the following prompts: 
x_m1= “A {feminine,masculine} person”, 
x_m2= “A person with {light, dark} skin”, 
x_m3 = “A person with {short, long} hair”, and
x_m4 = “A person in their {20, 50}s”.

Comment (lauraaisling) [reply papers/2312.14369_Quality-Diversity_Generative_Sampling_for_Learning_with_Synthetic_Data]: Incorporated all this in the additional details in PR.

**Pull request merged**